Switch away from deprecated v3 of the upload action.

oehmke opened the pull request:
As @billsacks said in his PR on the other repositories: 

 The v3 upload-artifact action is deprecated and will be removed in the
future (1/30/2025). This PR updates the action to use v4.

Note possibly breaking changes given in
https://github.com/actions/upload-artifact/blob/main/docs/MIGRATION.md.

I don't think that these apply because I believe the summarizer only runs one instance at a time, but let me know if you think that they do.  Thanks!

oehmke committed:
Switch away from deprecated v3 of the upload action.

billsacks commented:
Thanks @oehmke .

@danrosen25 also suggested changing the checkout action to use v4. I'll make that change and push it to this branch.

oehmke commented:
Thanks!

> On Jan 16, 2025, at 2:28 PM, Bill Sacks ***@***.***> wrote:
> 
> 
> Thanks @oehmke <https://github.com/oehmke> .
> 
> @danrosen25 <https://github.com/danrosen25> also suggested changing the checkout action to use v4. I'll make that change and push it to this branch.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/esmf-org/esmf-test-summary/pull/11#issuecomment-2596923278>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AE6A7U7M3U4N5VFHFGF2KP32LAQBPAVCNFSM6AAAAABVKOG2FCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKOJWHEZDGMRXHA>.
> You are receiving this because you were mentioned.
>

danrosen25 reviewed (changes requested):
There are several other outdated actions in this workflow file.

Also there's a non GitHub provided action that's receiving sensitive information. This needs to be removed entirely.
actions-js/push@master

Here's some guidance: https://github.com/esmf-org/esmf/commit/5d9527cb6b85a3748ca9c2a28b6ce585554e401d

billsacks committed:
Update various actions to their most recent versions

Inspired by comment from Dan Rosen:
https://github.com/esmf-org/esmf/pull/344#pullrequestreview-2557220055

billsacks commented:
@danrosen25 and @oehmke - I made the changes to versions, updating them all to their latest (thanks @danrosen25 !).

@danrosen25 - I looked through the commit you referenced but it isn't obvious to me how we need to change this push action. Is it clear to you?

oehmke commented:
Thanks Bill! 

> On Jan 16, 2025, at 2:41 PM, Bill Sacks ***@***.***> wrote:
> 
> 
> @danrosen25 <https://github.com/danrosen25> and @oehmke <https://github.com/oehmke> - I made the changes to versions, updating them all to their latest (thanks @danrosen25 <https://github.com/danrosen25> !).
> 
> @danrosen25 <https://github.com/danrosen25> - I looked through the commit you referenced but it isn't obvious to me how we need to change this push action. Is it clear to you?
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/esmf-org/esmf-test-summary/pull/11#issuecomment-2596949890>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AE6A7U5TCNR2RBYLAV66P232LARRVAVCNFSM6AAAAABVKOG2FCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKOJWHE2DSOBZGA>.
> You are receiving this because you were mentioned.
>

danrosen25 commented:
@billsacks @oehmke 
I'm not entirely sure and I won't have the time to look into this right now.

I don't know if this needs to work differently than the esmf to esmf-org.github.io connection because in this case the changes are being written to the same repository as the workflow.

If it is the same then a deploy key needs to be generated and used as a secret for pushing changes into the repository. A GitHub action can do anything it wants with an argument, including `github_token`. It can take an encryption key and save it to a server on the darkweb, you really don't know. The general rule is that you don't pass secrets into a third party action because of this vulnerability. We grant actions owned by GitHub a pass on this. These actions start with `actions`.  `actions-js` is an unknown organization

```
    - name: Checkout esmf-test-summary
      uses: actions/checkout@v4
      with:
        repository: esmf-org/esmf-test-summary
        path: esmf-test-summary
        ssh-key: ${{secrets.DEPLOY_KEY_STORED_FOR_ESMF_TEST_SUMMARY}}
        ref: main
```

```
    - name: Commit and push HTML
      run: | 
        cd ${{ github.workspace }}/esmf-test-summary
        git config user.name "github-actions[bot]"
        git config user.email "github-actions[bot]@users.noreply.github.com"
        git add .
        git diff-index --quiet HEAD || git commit --message "Publish Test Summary"
        git push origin
```

billsacks commented:
@danrosen25 - since the update of versions needs to happen within the next two weeks (which is when GitHub is permanently removing v3), how would you feel about merging the changes that are here so far and opening an issue to address the security issue later?

danrosen25 reviewed (approved):
Approved but please open an issue for removing action-js

billsacks commented:
> please open an issue for removing action-js

Opened as https://github.com/esmf-org/esmf-test-summary/issues/12

oehmke commented:
Thanks!

> On Jan 17, 2025, at 1:34 PM, Bill Sacks ***@***.***> wrote:
> 
> 
> Merged #11 <https://github.com/esmf-org/esmf-test-summary/pull/11> into main.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/esmf-org/esmf-test-summary/pull/11#event-15982058222>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AE6A7U4GHSTTZS2U377PCR32LFSNBAVCNFSM6AAAAABVKOG2FCVHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMJVHE4DEMBVHAZDEMQ>.
> You are receiving this because you were mentioned.
>
